sender.go

```diff
@@ -26,6 +31,10 @@ type Sender struct {
 // additional messages can be sent while the current goroutine is waiting
 // for the confirmation.
 func (s *Sender) Send(ctx context.Context, msg *Message) error {
+	if err := s.link.Check(); err != nil {
```

Comment:
so from a consumer point of view, the Check() func allows you to check periodically for the state of the link, but not to be notified of the link going bust unfortunately.

Reply:
Right, just a simple way to standardize on "is this link even good".

The other option (which I considered) was to provide "IsDetached" and "IsClosed" exports, but it seemed simpler to just take advantage of the predefined errors. 

In rhea (JS AMQP), for instance, they have these exported as 'is closed' and 'is remote closed' variables (well, and a specialized 'onDetached' handler).

> but not to be notified of the link going bust unfortunately.

For senders this isn't usually too bad (since you're basically always initiating an operation). I think your question is more about receivers. You (and I and @jhendrixMSFT) will definitely need to give that some thought!

Reply:
I was thinking of exposing an error channel, so that from the listener point of view, you can be notified as soon as this happens.

For example, for a sender that sends very few messages, say 1 message every few minutes. I'd want to know that the link died before I send, so that I can reconnect as soon as possible, and be healthy when the time comes to send a message.